inventory_management_system_api/services/catalogue_category.py

```diff
@@ -140,3 +153,36 @@ def update(
         return self._catalogue_category_repository.update(
             catalogue_category_id, CatalogueCategoryIn(**{**stored_catalogue_category.model_dump(), **update_data})
         )
+
+    def check_duplicate_property_names(self, properties: List[CatalogueItemPropertySchema], request: str):
+        """
+        Go through all the properties to check for any duplicate property names
+
+        :param properties: The supplied properties when creating or editing a catalogue category
+        :param request: Which request is being processed as the list of properties supplied have
+                        different types/format
+        :return: Returns true if any duplicate property names have been found
+        """
+
+        logger.info("Checking for duplicate property names")
+
+        list_of_names = []
+
+        if request == "post":
+            for dictionary in properties:
+                list_of_names.append(dictionary.name)
+
+        if request == "patch":
+            for dictionary in properties:
+                for key, value in dictionary.items():
+                    if key == "name":
+                        list_of_names.append(value)
+
+        unique_names = set()
+
+        for name in list_of_names:
+            if name.lower() in unique_names:
+                raise DuplicatePropertyName(
+                    f"Cannot have duplicate catalogue item property name: {name}"
+                )
+            unique_names.add(name.lower())
```

Comment:
A few things:
1. Lets refer to the properties as `catalogue_item_properties` at this level and keep it consistent with the rest of the module.
2. The method should be private as it's only accessed in this class so prefix it with `_`.
3. We no longer need the `request` arg if you pass `catalogue_category.catalogue_item_properties` to this method in the `update` method.
4. The method no longer returns so the type hint needs to be updated.
5. The docstring needs to be updated accordingly and also include `:raises ` info.
6. I don't think we need `list_of_names` as we can just loop through the `catalogue_item_properties`

```suggestion
    def _check_duplicate_catalogue_item_property_names(
        self, catalogue_item_properties: List[CatalogueItemPropertySchema]
    ) -> None:
        """
        Go through all the catalogue item properties to check for any duplicate names.

        :param catalogue_item_properties: The supplied catalogue item properties
        :raises DuplicateCatalogueItemPropertyName: If a duplicate catalogue item property name is found.
        """
        logger.info("Checking for duplicate catalogue item property names")
        seen_catalogue_item_property_names = set()
        for catalogue_item_property in catalogue_item_properties:
            catalogue_item_property_name = catalogue_item_property.name

            if catalogue_item_property_name.lower() in seen_catalogue_item_property_names:
                raise DuplicateCatalogueItemPropertyNameError(
                    f"Duplicate catalogue item property name: {catalogue_item_property_name}"
                )
            seen_catalogue_item_property_names.add(catalogue_item_property_name.lower())
```